targets/psa_switch/externs/psa_random.cpp

```diff
@@ -18,30 +18,45 @@
  */
 
 #include "psa_random.h"
-#include <bm/bm_sim/_assert.h>
+#include <bm/bm_sim/logger.h>
 
 #include <random>
 #include <thread>
+#include <iostream>
 
 namespace bm {
 
 namespace psa {
 
+void
+PSA_Random::init() {
+  valid_range = true;
+  min_val = min.get_uint64();
+  max_val = max.get_uint64();
+  range = max_val - min_val + 1;
+
+  if (range <= 0) {
+    valid_range = false;
+    Logger::get()->warn("Error: PSA extern random range must be positive.");
+  }
+
+  if ((range & (range - 1)) != 0) {
```

Comment:
I know the PSA specification says that implementations are allowed to impose restrictions like this, but is there any particular reason why the BMv2 implementation should?  It seems like without this `if` statement, the rest of the code would handle non-power-of-2 ranges just fine, yes?  If so, I'd recommend removing this `if` statement.

Reply:
The reason this restriction is mentioned in the PSA spec at all is that for some hardware targets, they might make such a restriction in order to make the hardware implementation simpler, e.g. generating N random bits, then adding them to min_value.

Reply:
That make sense, I will remove this if statement from bmv2 end :D thanks!

Reply:
Both should be fixed now: changed negative check to `!(min > max)` and removed the restriction that range must be a power of 2.